src/contracts/GPv2Settlement.sol

```diff
@@ -142,4 +143,41 @@ contract GPv2Settlement {
         );
         return IUniswapV2Pair(uint256(pairAddressBytes));
     }
+
+    /// @dev Execute a Uniswap trade exchanging the given amount of tokenIn
+    /// for the given amount of tokenOut.
+    ///
+    /// @param tokenIn The token sold on Uniswap.
+    /// @param tokenOut The token bought on Uniswap.
+    /// @param amountIn The amount sold.
+    /// @param amountOut The amount bought.
+    function uniswapTrade(
+        IERC20 tokenIn,
+        IERC20 tokenOut,
+        uint256 amountIn,
+        uint256 amountOut
+    ) internal {
+        bool depositToken0ReceiveToken1 = address(tokenIn) < address(tokenOut);
```

Comment:
Not sure where else we need this logic, but maybe it should be part of `uniswapPairAddress` as it seems tightly coupled to this logic?

Reply:
See https://github.com/gnosis/oba-contracts/pull/107#discussion_r512947426.

We need some way to encode the trade direction. Originally, I had proposed using sign of `d*`, but as @fedgiac pointed out, it might make more sense to use the ordering of the token hashes instead. The current idea is to modify the what we currently have (such as #89) to encode trade direction using the token ordering as is done here.